### PR TITLE
2 changes that i was having in my local copy for some time now

### PR DESCRIPTION
**Pull request by garbas**
*  passing NEO_* variables to webpack build process
*  loading babel-polyfill before any other code

### CODE REVIEW

File: config/webpack.core.js [hunk 1]
```diff
@@ -16,21 +16,25 @@ const SRC = path.dirname(SRC_FILE);
 const TESTS = path.join(CWD, 'tests');
 const USER_TEMPLATE = path.join(SRC, 'template.ejs');
 
+const NEO_VARIABLES = Object.keys(process.env)
+  .filter(x => x.toUpperCase().startsWith('NEO_'))
+  .reduce((r, x) => { r['process.env.' + x] = JSON.stringify(process.env[x]); return r; }, {});
```
Comment (eliperelman) [inline config/webpack.core.js]: Can we 2-line this, and template variable?

```js
.reduce((r, x) => {
  r[`process.env.${x}`] = JSON.stringify(process.env[x]);
  return r;
}, {});
```

File: config/webpack.core.js [hunk 2]
```diff
@@ -16,21 +16,25 @@ const SRC = path.dirname(SRC_FILE);
...
+
 let loader = name => `${name}?${qs.stringify(require(`.\/${name}`), {
   encode: false,
   arrayFormat: 'brackets'
 })}`;
 
 module.exports = {
-  entry: [SRC_FILE],
+  entry: ['babel-polyfill', SRC_FILE],
```
Comment (eliperelman) [inline config/webpack.core.js]: Can you also tell me what the `babel-polyfill` here does?

Comment (garbas) [inline config/webpack.core.js]: ``babel-polyfill`` is needed for support of generators which are used in ``redux-saga``. it needs to be added before everything else.

Comment (eliperelman) [inline config/webpack.core.js]: Oh I see. Then in that case, we should probably instead modify babel.js to this:

```js
module.exports = {
  plugins: [
    require.resolve('babel-plugin-transform-runtime')
  ],
  presets: [
    require.resolve('babel-preset-es2015'),
    require.resolve('babel-preset-stage-0'),
    require.resolve('babel-preset-react')
  ]
};
```

Would that work the same?

Comment (garbas) [inline config/webpack.core.js]: i'm not sure, but this what it was advised to do on when integrating with webpack

**Pull request merged**